### ISSUE TITLE
Problem with intersection types

### ISSUE THREAD

**Issue by jarlah** [facebook/flow#4083]
I have seen some wierd behaviour with the flow type checker. It seems that accessing fields that does not exist on an intersection type, incorrectly passes flow type checking.
```
type A = { 
  x: string
}
type B = {
  y: string
}
type C = A & B
const cObj: C =  ....
console.log(cObj.fieldThatDoesNotExist); // Passes flow checking, why??
const aObj:  A  = c
console.log(aObj.fieldThatDoesNotExist); // Correctly fails flow checking! Because field is not defined in A
```

**Comment by leviroth** [comment]
I couldn't reproduce this:

```
type A = {
  x: string
}
type B = {
  y: string
}
type C = A & B
const cObj: C = {x: "foo", y: "bar"}
console.log(cObj.fieldThatDoesNotExist);
const aObj:  A  = {x: "foo"}
console.log(aObj.fieldThatDoesNotExist);
```

On the first `console.log` line, I get ``property `fieldThatDoesNotExist`. Property cannot be accessed on any member of intersection type``.

**Comment by jarlah** [comment]
i had 0.46. was fixed in 0.47

0.47.0

Likely to cause new Flow errors:

We are now sealing the object type that is passed to a React component's props. This means Flow will start complaining if you try and access unknown React props.
